dot/digest/digest.go

```diff
@@ -395,14 +391,13 @@ func (h *Handler) handleResume(d *types.ConsensusDigest) error {
 		return err
 	}
 
-	p := &types.GrandpaResume{}
-	dec, err := scale.Decode(d.Data[1:], p)
+	var r types.GrandpaResume
+	err = scale.Unmarshal(d.Data[1:], &r)
```

Comment:
Yeah I think there is more room for refactor. It does this because consensus digest defines the first byte as one of:
`var (
	GrandpaScheduledChangeType = byte(1)
	GrandpaForcedChangeType    = byte(2)
	GrandpaOnDisabledType      = byte(3)
	GrandpaPauseType           = byte(4)
	GrandpaResumeType          = byte(5)
)`
Maybe make these VDTs?

Reply:
Yes these are all rust enums and therefore should be VDTs.